python/pyarrow/types.py

```diff
@@ -20,295 +20,54 @@
-def is_nested(t):
-    return t.id in _NESTED_TYPES
-
-
-@doc(is_null, datatype="run-end encoded")
-def is_run_end_encoded(t):
-    return t.id == lib.Type_RUN_END_ENCODED
-
-
-@doc(is_null, datatype="date, time, timestamp or duration")
-def is_temporal(t):
-    return t.id in _TEMPORAL_TYPES
-
-
-@doc(is_null, datatype="timestamp")
-def is_timestamp(t):
-    return t.id == lib.Type_TIMESTAMP
-
-
-@doc(is_null, datatype="duration")
-def is_duration(t):
-    return t.id == lib.Type_DURATION
-
-
-@doc(is_null, datatype="time")
-def is_time(t):
-    return t.id in _TIME_TYPES
-
-
-@doc(is_null, datatype="time32")
-def is_time32(t):
-    return t.id == lib.Type_TIME32
-
-
-@doc(is_null, datatype="time64")
-def is_time64(t):
-    return t.id == lib.Type_TIME64
-
-
-@doc(is_null, datatype="variable-length binary")
-def is_binary(t):
-    return t.id == lib.Type_BINARY
-
-
-@doc(is_null, datatype="large variable-length binary")
-def is_large_binary(t):
-    return t.id == lib.Type_LARGE_BINARY
-
-
-@doc(method="is_string")
-def is_unicode(t):
-    """
-    Alias for {method}.
-
-    Parameters
-    ----------
-    t : DataType
-    """
-    return is_string(t)
-
-
-@doc(is_null, datatype="string (utf8 unicode)")
-def is_string(t):
-    return t.id == lib.Type_STRING
-
-
-@doc(is_unicode, method="is_large_string")
-def is_large_unicode(t):
-    return is_large_string(t)
-
-
-@doc(is_null, datatype="large string (utf8 unicode)")
-def is_large_string(t):
-    return t.id == lib.Type_LARGE_STRING
-
-
-@doc(is_null, datatype="fixed size binary")
-def is_fixed_size_binary(t):
-    return t.id == lib.Type_FIXED_SIZE_BINARY
-
-
-@doc(is_null, datatype="variable-length binary view")
-def is_binary_view(t):
-    return t.id == lib.Type_BINARY_VIEW
-
-
-@doc(is_null, datatype="variable-length string (utf-8) view")
-def is_string_view(t):
-    return t.id == lib.Type_STRING_VIEW
-
-
-@doc(is_null, datatype="date")
-def is_date(t):
-    return t.id in _DATE_TYPES
-
-
-@doc(is_null, datatype="date32 (days)")
-def is_date32(t):
-    return t.id == lib.Type_DATE32
-
-
-@doc(is_null, datatype="date64 (milliseconds)")
-def is_date64(t):
-    return t.id == lib.Type_DATE64
-
-
-@doc(is_null, datatype="map")
-def is_map(t):
-    return t.id == lib.Type_MAP
-
-
-@doc(is_null, datatype="decimal")
-def is_decimal(t):
-    return t.id in _DECIMAL_TYPES
-
-
-@doc(is_null, datatype="decimal128")
-def is_decimal128(t):
-    return t.id == lib.Type_DECIMAL128
-
-
-@doc(is_null, datatype="decimal256")
-def is_decimal256(t):
-    return t.id == lib.Type_DECIMAL256
-
-
-@doc(is_null, datatype="dictionary-encoded")
-def is_dictionary(t):
-    return t.id == lib.Type_DICTIONARY
-
-
-@doc(is_null, datatype="interval")
-def is_interval(t):
-    return t.id == lib.Type_INTERVAL_MONTH_DAY_NANO
-
-
-@doc(is_null, datatype="primitive type")
-def is_primitive(t):
-    return lib._is_primitive(t.id)
+                         is_float_value,
+                         ensure_type,
```

Comment:
Despite not beginning with a underscore, this is not a public function, so we should not export it.

Reply:
fix it.